### ISSUE TITLE
Mutations fail when updating to falsy values

### ISSUE THREAD

**Issue by zchsh** [marmelab/aor-graphql#39]
> **NOTE**
> After getting this working at least for my use case, I realized that the [aor-graphql-client-graphcool package on npm](https://www.npmjs.com/package/aor-graphql-client-graphcool) that I'm using points to a repo that's old and gone. So I wasn't sure where I should post this - but it seems to apply to this repo as well!

Heya!

I'm trying to set up admin-on-rest with a graph.cool instance, with the npm package `aor-graphql-client-graphcool`, but updating Booleans to false and other fields to falsy values doesn't work - instead, those values are excluded from the query.

I think this is because variables are being filtered out in `buildGqlQuery.js`, where the variable `validVariables` is created.

> (similar filtering happens in two other places in `queryBuilder.js`, but I think the way I have AoR setup means those instances don't seem to get used)

I'm not sure if this fix works in every case, but in the functions `buildArgs` and `buildApolloArgs` in `buildGqlQuery.js`, I've just replaced the below:

```
const validVariables = Object.keys(variables).filter(k => !!variables[k] && variables[k] !== null);
```

with:

```
const validVariables =  Object.keys(variables).filter(k => typeof variables[k] !== 'undefined' && variables[k] !== null);
```

> (in the npm package it's not quite the same hotshot arrow syntax, but close enough!)

**Comment by Toosick** [comment]
My scenario is similar, but I needed to make some similar changes.
___
## Change 1
`const validVariables = Object.keys(variables).filter(k => !!variables[k] && variables[k] !== null);`
with
`const validVariables =  Object.keys(variables).filter(k => typeof variables[k] !== 'undefined');`

## Change 2

```
return { ...acc, [`$${arg.name}`]: 'ID!' };
```
 with 
```
return { ...acc, [`$${arg.name}`]: 'ID' };`
```
___
**Use case:** trying to unset a 1 to 1 relation with null.

Ideally change 2 should be done based on the kind of the argument. Ex: id has a NON_NULL kind and our custom 1 to 1 relation field has a kind of SCALAR.

**Comment by djhi** [comment]
Fixed by #50
